cmd/thanos/sidecar.go

```diff
@@ -363,6 +364,9 @@ func runSidecar(
 				uploadCompactedFunc, conf.shipper.allowOutOfOrderUpload, metadata.HashFunc(conf.shipper.hashFunc), conf.shipper.metaFileName)
 
 			return runutil.Repeat(30*time.Second, ctx.Done(), func() error {
+				// Generate random delay using upload jitter.
+				jitter := time.Duration(rand.Int63n(int64(conf.shipper.uploadJitter)))
+				time.Sleep(jitter)
```

Comment:
Instead of sleeping, can we do something similar to this https://github.com/cortexproject/cortex/blob/master/pkg/util/time.go#L73?

I image we will add a new function called `runutil.RepeatWithJitter` which takes the same parameters as `runutil.Repeat` but also takes `variancePerc`.
We can just hardcode `variancePerc` to a sane value like 0.05 so allows [-1.5s, 1.5s] jitter.